2024Robot_Team3484/src/main/cpp/subsystems/SwerveModule.cpp

```diff
@@ -0,0 +1,127 @@
+#include "subsystems/SwerveModule.h"
+
+#include <units/length.h>
+#include <units/velocity.h>
+#include <units/angle.h>
+#include <units/angular_velocity.h>
+#include <units/voltage.h>
+
+#include <frc/geometry/Rotation2d.h>
+
+using namespace frc;
+using namespace units;
+using namespace ctre::phoenix6;
+using namespace SwerveConstants::DrivetrainConstants;
+
+SwerveModule::SwerveModule(const int module_location) 
+        : _drive_motor(DRIVE_MOTOR_PORTS[module_location]),
+          _steer_motor(STEER_MOTOR_PORTS[module_location]),
+          _steer_encoder(ENCODER_PORTS[module_location])
```

Comment:
I suggest considering passing values such as CAN ID's, IO channels, and configuration values through the constructor instead of referencing the robot-specific constants directly in at the subsystem level (or sub-subsystem level in this case). This helps the code be more portable year-to-year (as well as making it easier to add these pieces of code to the team's library).

It can help clean up the declarations by defining `struct`s that contain related pieces of information. In the case here, you may create a structure that has values for current limiting, as well as the CAN ID of the motor in question. e.g.:
```
typedef struct
{
    int CAN_ID;
    bool IsReversed;
    int Current_Thresh;
    int Current_Limit;
    int Current_Time;
} SC_SwerveModule_Cfg;
```

Side note: There already is a struct called `SC_SwerveModuleConfig` in `SC_Datatypes.h`. Feel free to change it however you want.